build/codegen/codegen.sh

```diff
@@ -35,7 +35,7 @@ bash ${codegendir}/generate-groups.sh \
     "${GROUP_VERSIONS}" \
     --output-base "${scriptdir}/../../vendor" \
     --go-header-file "${scriptdir}/boilerplate.go.txt"
-cp -r "${scriptdir}/../../vendor/github.com/rook/rook/pkg/" "${scriptdir}/../../"
+cp -r "${scriptdir}/../../vendor/github.com/rook/rook/pkg/" "${scriptdir}/../../pkg/"
```

Comment:
```suggestion
cp -a "${scriptdir}/../../vendor/github.com/rook/rook/pkg/" "${scriptdir}/../../"
```

Reply:
Nope this doesn't fix it. Why would the version of `cp` make a difference here? By removing `pkg` from the target path, it is telling the command to copy up one level in the directory structure, right? 

This script hasn't changed for a long time unless it was related to the go modules change. What is the issue that you were seeing that required the change on your side?

Reply:
Doing `cp -r` tell `cp` to recursively copy `"${scriptdir}/../../vendor/github.com/rook/rook/pkg/" ` into `"${scriptdir}/../../pkg/"` which ends up copying `pkg` too, where it should only copy the its content. Currently, we end up with `pkg/pkg/` being added and we don't want that.

The `cp` version on OSX is old and the behavior is different than the one in gnu-tools. I'm surprised to see that `cp -a` doesn't work. What is it doing?

Reply:
> The cp version on OSX is old and the behavior is different than the one in gnu-tools. I'm surprised to see that cp -a doesn't work. What is it doing?

`cp -a` is working the same for me as `cp -r`. i'll experiment more with this in a bit...

> Also, cp was added recently with #5110.

ah, I missed that change

Reply:
@leseb simply removing the trailing backslash on the source path fixed the issue for me. Does that work for you?

Reply:
That works too!